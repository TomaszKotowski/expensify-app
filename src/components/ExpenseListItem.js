import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`} >
            <h3>{description}</h3>
        </Link>
        <p>Amount: {amount}</p>
        <p>Created at: {createdAt}</p>
    </div>
);

export default ExpenseListItem;