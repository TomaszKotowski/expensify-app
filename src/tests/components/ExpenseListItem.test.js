import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render ExpenseList with expenses', () => {
    const wrapper = shallow(
        <ExpenseListItem
            id={expenses[0].id}
            description={expenses[0].description}
            amount={expenses[0].amount}
            createdAt={expenses[0].createdAt}
        />);
    expect(wrapper).toMatchSnapshot();
});