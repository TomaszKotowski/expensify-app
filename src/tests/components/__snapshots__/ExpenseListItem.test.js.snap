// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    Amount: 
    195
  </p>
  <p>
    Created at: 
    0
  </p>
</div>
`;
